### PR TITLE
Update target resource name

### PR DESCRIPTION
**Pull request by wangjiejacques**
Resolves https://github.com/tuist/tuist/issues/4540
Request for comments document (if applies):

### Short description 📝

> Update target resource name

### How to test the changes locally 🧐

> Acceptance tests and unit tests should be enough. 

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, whenever it should be included in the “Added”, “Fixed” or “Changed” section of the CHANGELOG. Note: when included in the CHANGELOG, the title of the PR will be used as entry, please make sure it is clear and suitable.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.
- [x] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

**Review by fortmarek — APPROVED** [review]
Given the bundle name should be an internal detail, I don't mind adhering to SPM's convention. Would wait for @kwridan with review as well if he can think of something that this change would break.

File: Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift
Comment (danyf90) [inline Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift:166]: why empty? 🤔

Comment (wangjiejacques) [reply Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift:166]: The test shall pass with any value, so this value is irrelevant to this test.

Maybe you have a better idea, or we can keep the original value?

Comment (danyf90) [reply Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift:166]: Aren't we checking the file content 2 lines below? how is it not relevant? 🤔

Comment (wangjiejacques) [reply Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift:166]: The `bundleName` is only used `if !target.supportsResources`, in this test, the target is a `framework`, which supports Resources.

Comment (wangjiejacques) [reply Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift:166]: Here is the code of `ResourcesProjectMapper.fileContent`
```swift
    // swiftlint:disable:next function_body_length
    static func fileContent(targetName: String, bundleName: String, target: Target) -> String {
        if !target.supportsResources {
            return """
            // Bundle name is used here
            // ...

            """
        } else {
            return """
            // Bundle name is not used here.
            // ...
            """
        }
    }

```

Comment (danyf90) [reply Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift:166]: So if you use `.framework` instead of `staticFramework` for a dependency (you can map it in Dependencies.swift, or Tuist could chose to use it according to the mapping logic), then the bundle name won't match the SPM one?

cc @fortmarek

**Review by danyf90 — CHANGES_REQUESTED** [review]
Would like to clarify [this](https://github.com/tuist/tuist/pull/4542#discussion_r895049761) before merging

File: Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift
Comment (wangjiejacques) [reply Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift:166]: > then the bundle name won't match the SPM one?

If it's framework, then we don't need this bundle name.
Check this [comment](https://github.com/tuist/tuist/blob/685840668d0a63b2e9d22fda753c9c0e1abd85c1/Sources/TuistGenerator/Mappers/ResourcesProjectMapper.swift#L144).

> /// Since \(targetName) is a \(target.product), the bundle for classes within this module can be used directly.

Comment (wangjiejacques) [reply Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift:166]: Hello @fortmarek @kwridan , can you take a look ?

Comment (fortmarek) [reply Tests/TuistGeneratorTests/ProjectMappers/ResourcesProjectMapperTests.swift:166]: Yeah, I believe we don't create a bundle if it's a `framework`, isn't that correct?

**Review by danyf90 — APPROVED** [review]

**Comment by wangjiejacques** [comment]
> Thanks for contributing this @wangjiejacques
> 
> The proposed changes seem reasonable, that said reading through the original issue I worry that depending on the bundle names generated by Tuist directly rather than accessing them via the API may be flaky _(for example if anyone already depended on the current naming scheme `<Target>Resources` they will be in the same boat!)_

I agree with you, our current solution that depending on the bundle names is not ideal, but as Swift Package Manager has no official way to Distributing Binary Static Frameworks with bundle resources, this is the only solution.

FYI, I have discussed this with engineers from apple, they confirmed this is the only way to do so.

**Pull request merged**
